test/integration/rest-basic/index.spec.ts

```diff
@@ -30,8 +31,19 @@ test('functions basics', async ({ page }) => {
   const runtimeModel = new ToolpadRuntime(page);
   await runtimeModel.gotoPage(app.id, 'page1');
 
-  await page.locator('text="query1: query1_value"').waitFor({ state: 'visible' });
-  await page.locator('text="query2: undefined"').waitFor({ state: 'visible' });
+  await expect(page.locator('text="query1: query1_value"')).toBeVisible();
+  await expect(page.locator('text="query2: undefined"')).toBeVisible();
   await page.locator('button:has-text("fetch query2")').click();
-  await page.locator('text="query2: query2_value"').waitFor({ state: 'visible' });
+  await expect(page.locator('text="query2: query2_value"')).toBeVisible();
+  await expect(page.locator('text="browserQuery: browserQuery_value"')).toBeVisible();
+  await expect(page.locator('text="browserQuery: browserQuery_value"')).toBeVisible();
+
+  const editorModel = new ToolpadEditor(page, browserName);
+  await editorModel.goto(app.id);
+
+  await editorModel.componentEditor.getByRole('button', { name: 'browserQuery' }).click();
+  const queryEditor = page.getByRole('dialog', { name: 'browserQuery' });
```

Comment:
Should we create an abstraction for testing preview queries like this?

Reply:
We can, I was a bit hesitant to abstract it while having only one use-case so far

Reply:
Yeah, just a comment but probably not really necessary for now.